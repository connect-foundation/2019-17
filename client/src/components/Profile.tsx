import React from 'react';
import styled from 'styled-components';
import { DEFAULT } from 'constant';

const Image = styled.img<{ size: string }>`
  width: ${props => props.size};
  height: ${props => props.size};
  border-radius: 50%;
`;

interface IProps {
  imageUrl: string;
  size: string;
  alt: string;
  className?: string;
}

function Profile({ imageUrl, size, alt, className }: IProps) {
  return <Image className={className} src={imageUrl} size={size} alt={alt} />;
}

Profile.defaultProps = {
  size: '40px',
  alt: 'profile image',
  imageUrl: DEFAULT.PROFILE
};

export default Profile;
