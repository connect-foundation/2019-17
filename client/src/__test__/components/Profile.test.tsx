import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import Profile from 'components/Profile';

afterEach(cleanup);

test('<Profile />', () => {
  const { container, getByAltText, rerender } = render(<Profile />);
  expect(container).toMatchSnapshot();
  rerender(<Profile alt={'feed profile'} size={"40px"}/>);
  expect(getByAltText('feed profile')).toBeInTheDocument();
});
