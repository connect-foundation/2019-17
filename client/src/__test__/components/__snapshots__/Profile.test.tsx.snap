// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> 1`] = `
<div>
  <img
    alt="profile image"
    class="sc-bdVaJa biickM"
    src="/images/profile.jpg"
  />
</div>
`;
